File: claims-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__ClaimsProviderDescriptor.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.claims.ClaimsProvider;
import se.curity.identityserver.sdk.plugin.descriptor.ClaimsProviderPluginDescriptor;

public final class ${pluginName}ClaimsProviderDescriptor implements ClaimsProviderPluginDescriptor<${pluginName}ClaimsProviderConfig>
{
    @Override
    public Class<? extends ClaimsProvider> getClaimsProvider()
    {
        return ${pluginName}ClaimsProvider.class;
    }

    @Override
    public String getPluginImplementationType()
    {
        return "${pluginName.toLowerCase()}";
    }

    @Override
    public Class<? extends ${pluginName}ClaimsProviderConfig> getConfigurationType()
    {
        return ${pluginName}ClaimsProviderConfig.class;
    }
}

File: kotlin-claims-provider-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__ClaimsProviderDescriptor.kt
package ${package}.${pluginName}

import se.curity.identityserver.sdk.plugin.descriptor.ClaimsProviderPluginDescriptor

class ${pluginName}ClaimsProviderDescriptor: ClaimsProviderPluginDescriptor<${pluginName}ClaimsProviderConfig>
{
    override fun getClaimsProvider() = ${pluginName}ClaimsProvider::class.java

    override fun getPluginImplementationType() = "${pluginName.toLowerCase()}"

    override fun getConfigurationType() = ${pluginName}ClaimsProviderConfig::class.java
}
